src/sentry/uptime/subscriptions/tasks.py

```diff
@@ -0,0 +1,137 @@
+
+
+@instrumented_task(
+    name="sentry.uptime.subscriptions.tasks.create_uptime_subscription",
+    queue="uptime",
+    default_retry_delay=5,
+    max_retries=5,
+)
+def create_remote_uptime_subscription(uptime_subscription_id, **kwargs):
+    try:
+        subscription = UptimeSubscription.objects.get(id=uptime_subscription_id)
+    except UptimeSubscription.DoesNotExist:
+        metrics.incr("uptime.subscriptions.create.subscription_does_not_exist")
+        return
+    if subscription.status != UptimeSubscription.Status.CREATING.value:
+        metrics.incr("uptime.subscriptions.create.incorrect_status")
+        return
+
+    subscription_id = send_uptime_subscription_config(subscription)
+    # TODO: Ideally this should actually be `PENDING_FIRST_UPDATE` so we can validate it's really working as expected
+    subscription.update(
+        status=QuerySubscription.Status.ACTIVE.value, subscription_id=subscription_id
+    )
+
+
+@instrumented_task(
+    name="sentry.uptime.subscriptions.tasks.delete_uptime_subscription",
+    queue="uptime",
+    default_retry_delay=5,
+    max_retries=5,
+)
+def delete_remote_uptime_subscription(uptime_subscription_id, **kwargs):
+    try:
+        subscription = UptimeSubscription.objects.get(id=uptime_subscription_id)
+    except UptimeSubscription.DoesNotExist:
+        metrics.incr("uptime.subscriptions.delete.subscription_does_not_exist")
+        return
+
+    if subscription.status not in [
+        UptimeSubscription.Status.DELETING.value,
+        UptimeSubscription.Status.DISABLED.value,
+    ]:
+        metrics.incr("uptime.subscriptions.delete.incorrect_status")
+        return
+
+    subscription_id = subscription.subscription_id
+    if subscription.status == QuerySubscription.Status.DELETING.value:
+        subscription.delete()
+    else:
+        subscription.update(subscription_id=None)
+
+    if subscription_id is not None:
+        send_uptime_config_deletion(subscription.subscription_id)
+
+
+def send_uptime_subscription_config(subscription: UptimeSubscription) -> str:
+    # Whenever we create/update a config we always want to generate a new subscription id. This allows us to validate
+    # that the config took effect
+    subscription_id = uuid4().hex
+    check_config = uptime_subscription_to_check_config(subscription, subscription_id)
+    send_uptime_config_message(subscription_id, _get_config_codec().encode(check_config))
+    return subscription_id
+
+
+def uptime_subscription_to_check_config(
+    subscription: UptimeSubscription, subscription_id: str
+) -> CheckConfig:
+    return {
+        "subscription_id": subscription_id,
+        "url": subscription.url,
+        "interval_seconds": subscription.interval_seconds,  # type: ignore[typeddict-item]
+        "timeout_ms": subscription.timeout_ms,
+    }
+
+
+def send_uptime_config_deletion(subscription_id: str) -> None:
+    send_uptime_config_message(subscription_id, b"")
```

Comment:
It's actually not empty bytes, it's `None`. The types in arroyo are wrong though

Reply:
Can be replaced with the stuff in `uptime.config_producer` anyway